lib/sdk/server/src/main/java/com/launchdarkly/sdk/server/integrations/FileDataSourceParsing.java

```diff
@@ -72,14 +72,18 @@ static final class FlagFileRep {
   }
 
   static abstract class FlagFileParser {
-    private static final FlagFileParser jsonParser = new JsonFlagFileParser();
-    private static final FlagFileParser yamlParser = new YamlFlagFileParser();
+    static class JsonParserHolder {
+      static final FlagFileParser INSTANCE = new JsonFlagFileParser();
+    }
+    static class YamlParserHolder {
+      static final FlagFileParser INSTANCE = new YamlFlagFileParser();
```

Comment:
should be private with a getter so it only gets initialized on access as opposed to having direct reference to variable.

Reply:
you mean something like this?

```suggestion
      private static final FlagFileParser INSTANCE = new YamlFlagFileParser();
      static final FlagFileParser getInstance() {
        return INSTANCE;
      }
```